index.js

```diff
@@ -88,11 +95,24 @@ function retryRequest(requestOpts, opts, callback) {
       delayStream = through({ objectMode: opts.objectMode });
       requestStream = opts.request(requestOpts);
 
-      requestStream
-        .on('error', onResponse)
-        .on('response', onResponse.bind(null, null))
-        .on('complete', retryStream.emit.bind(retryStream, 'complete'))
-        .pipe(delayStream);
+      var events = {
+        error: onResponse,
+        response: onResponse.bind(null, null),
+        complete: retryStream.emit.bind(retryStream, 'complete')
+      };
+
+      for (var eventName in events) {
+        requestStream.once(eventName, events[eventName]);
+      }
+
+      removeAllListeners = function() {
+        for (var eventName in events) {
+          requestStream.on(eventName, function() {});
```

Comment:
Strange. This seems to be coming from the `pipe` call.

```
internal/streams/legacy.js:59
      throw er; // Unhandled stream error in pipe.
```

It looks like `pipe` binds its own error handler on the source stream that checks to make sure there are error handlers, but by the time its handler fires this one is gone. The remove might need to happen on the next tick.